### PR TITLE
Migrate async/future_stream to non-null by default

### PR DESCRIPTION
**Pull request by cbracken**
Partial fix for #606.

### CODE REVIEW

File: lib/src/async/future_stream.dart [hunk 1]
```diff
@@ -56,31 +54,31 @@ class FutureStream<T> extends Stream<T> {
...
-      _subscription = stream.listen(_controller.add,
-          onError: _controller.addError, onDone: _controller.close);
+      _subscription = stream.listen(_controller!.add,
+          onError: _controller!.addError, onDone: _controller!.close);
     });
   }
 
   void _onCancel() {
-    if (_subscription != null) _subscription.cancel();
+    if (_subscription != null) _subscription?.cancel();
```
Comment (jtmcdole) [inline lib/src/async/future_stream.dart:70]: Why even do the null test when you can just make this:

`_subscription?.cancel();`

Comment (cbracken) [reply lib/src/async/future_stream.dart]: I'll go hang my head in shame now. I bet I did that in stream_router too.

Comment (cbracken) [reply lib/src/async/future_stream.dart]: Done.

Comment (jtmcdole) [reply lib/src/async/future_stream.dart]: <3

File: lib/src/async/future_stream.dart [hunk 2]
```diff
@@ -39,8 +37,8 @@ class FutureStream<T> extends Stream<T> {
       // Since [controller] is synchronous, it's likely that emitting an error
       // will cause it to be cancelled before we call close.
       if (_controller != null) {
-        _controller.addError(e, stackTrace);
-        _controller.close();
+        _controller?.addError(e, stackTrace);
```
Comment (jtmcdole) [inline lib/src/async/future_stream.dart:40]: also doesn't need the surrounding check; but then again, there are two?  Would this be the case of `_controller!.addError()..close()`?

**Review by jtmcdole — APPROVED** [review]

File: lib/src/async/future_stream.dart (hunk 2 above)
Comment (cbracken) [reply lib/src/async/future_stream.dart:40]: You could do `_controller!..addError()..close();` but dartfmt will reformat to a three-liner. Given these have null-checks though I can kill off the conditional around this though.

Comment (cbracken) [reply lib/src/async/future_stream.dart:40]: Actually I'm not sure there's a one-liner that's possible regardless of dartfmt. You could imagine that addError were implemented such that it had access to enough context to null out _controller, so a second bang is necessary, but I don't think is supported in the language.

**Pull request merged**